python/lsst/ip/diffim/imageMapReduce.py

```diff
@@ -275,15 +276,36 @@ def run(self, mapperResults, exposure, **kwargs):
                     wtsView = afwImage.ImageI(weights, item.getBBox())
                     wtsView.getArray()[isNotNan] += 1
 
+        isNan = (np.isnan(newMI.getImage().getArray()) |
+                  np.isnan(newMI.getVariance().getArray()))
+        if np.sum(isNan) > 0:
+            # set mask to INVALID for pixels where produced exposure is NaN
+            mask = newMI.getMask()
+            # New mask plane - for debugging map-reduced images
+            mask.addMaskPlane('INVALID_MAPREDUCE')
+            bad = mask.getPlaneBitMask(['INVALID_MAPREDUCE', 'BAD', 'NO_DATA'])
+            mask.getArray()[isNan] |= bad
+
         if reduceOp == 'average':
             wts = weights.getArray().astype(np.float)
             self.log.info('AVERAGE: Maximum overlap: %f', wts.max())
             self.log.info('AVERAGE: Average overlap: %f', np.nanmean(wts))
+            self.log.info('AVERAGE: Minimum overlap: %f', wts.min())
+            wtsZero = wts == 0.
+            self.log.info('AVERAGE: Number of zero pixels: %f (%f%%)', np.sum(wtsZero),
+                          np.mean(wtsZero) * 100.)
+            wts[wtsZero] = 1e-29  # avoid division by zero warnings -- will set to nan and flag them below
             newMI.getImage().getArray()[:, :] /= wts
             newMI.getVariance().getArray()[:, :] /= wts
-            wtsZero = wts == 0.
-            newMI.getImage().getArray()[wtsZero] = newMI.getVariance().getArray()[wtsZero] = np.nan
-            # TBD: set mask to something for pixels where wts == 0. Shouldn't happen. (DM-10009)
+            if np.sum(wtsZero) > 0:
+                newMI.getImage().getArray()[wtsZero] = newMI.getVariance().getArray()[wtsZero] = np.nan
```

Comment:
why are you setting these values to nan here? is that not redundant info with the mask plane bits set? Do we ever want nans in an image? Doesn't that make things like afw statistics need to do more work?